### PR TITLE
Fix: squid:S1854, Dead stores should be removed

### PR DESCRIPTION
**Pull request by elkfrawy-df**
This pull request is focused on resolving occurrences of Sonar rule 
 squid:S1854 - “ Dead stores should be removed ”. 
 You can find more information about the issue here: 
 https://dev.eclipse.org/sonar/rules/show/squid:S1854

 Please let me know if you have any questions.
Ayman Elkfrawy.

### CODE REVIEW

File: core/src/main/java/smile/clustering/CLARANS.java [hunk 1]
```diff
@@ -230,7 +230,7 @@ private double getRandomNeighbor(T[] data, T[] medoids, int[] y, double[] d) {
         int n = data.length;
 
         int index = Math.randomInt(k);
-        T medoid = null;
+        T medoid;
```
Comment (haifengl) [inline core/src/main/java/smile/clustering/CLARANS.java]: This is actually to remove warning like "uninitialized variable". Maybe today javac doesn't report that warning any more?

File: data/src/main/java/smile/data/parser/ArffParser.java [hunk 2]
```diff
@@ -183,7 +183,7 @@ private void getNextToken(StreamTokenizer tokenizer) throws IOException, ParseEx
      */
     private String readHeader(StreamTokenizer tokenizer, List<Attribute> attributes) throws IOException, ParseException {
         /// The name of dataset.
-        String relationName = null;
+        String relationName;
```
Comment (haifengl) [inline data/src/main/java/smile/data/parser/ArffParser.java]: same as above

File: graph/src/test/java/smile/graph/AdjacencyListTest.java [hunk 3]
```diff
@@ -349,8 +349,6 @@ public void testDfs() {
     @Test
     public void testDfs2() {
         System.out.println("dfs connected component");
```
Comment (haifengl) [inline graph/src/test/java/smile/graph/AdjacencyListTest.java]: Please keep these variables. I may miss some test on these results. I should add more tests, not remove them. Thanks!

File: graph/src/test/java/smile/graph/AdjacencyListTest.java [hunk 4]
```diff
@@ -423,8 +421,6 @@ public void testBfs() {
     @Test
     public void testBfs2() {
         System.out.println("bfs connected component");
-        int[] size = {3, 5};
```
Comment (haifengl) [inline graph/src/test/java/smile/graph/AdjacencyListTest.java]: same as above

File: graph/src/test/java/smile/graph/AdjacencyMatrixTest.java [hunk 5]
```diff
@@ -348,8 +348,6 @@ public void testDfs() {
     @Test
     public void testDfs2() {
         System.out.println("dfs connected component");
-        int[] size = {3, 5};
```
Comment (haifengl) [inline graph/src/test/java/smile/graph/AdjacencyMatrixTest.java]: same as above

**Comment by haifengl** [comment]
I think we should withdraw this one. These dead stores are not really errors and have no performance hit. They were used on purpose before to make javac happy. Otherwise, we will get a lot of warnings.

**Comment by elkfrawy-df** [comment]
@haifengl This is more for readability purposes, that there is no meaning for the initial value and it might confuse the developer.

File: plot/src/main/java/smile/swing/FontChooser.java [hunk 6]
```diff
@@ -484,7 +483,7 @@ public TextFieldKeyHandlerForListSelectionUpDown(JList<String> list) {
 
         @Override
         public void keyPressed(KeyEvent e) {
-            int i = targetList.getSelectedIndex();
+            int i;
             switch (e.getKeyCode()) {
```
Comment (elkfrawy-df) [inline plot/src/main/java/smile/swing/FontChooser.java]: This assignment for example is redundant and has no effect which reduces the readability.

**Comment by haifengl** [comment]
Without them, I will get a lot of warnings. Have you check build.sbt on javac options. I guess that your compiler settings are different.

**Comment by haifengl** [comment]
I updated FontChooser by removing i. Please have a look.

**Comment by haifengl** [comment]
Please remove the change to FontChooser. I will merge the rest changes. Thanks!

**Comment by haifengl** [comment]
I understand these assignments are useless in Java because the language does it implicitly. However, I prefer to keep them in case that I may port the code to c/c++ and/or cuda. If so, I can simply copy the code without worries I miss initializations.

**Comment by elkfrawy-df** [comment]
That is fine, no problem.

**Pull request closed without merging**